src/jit/morph.cpp

```diff
@@ -2951,7 +2955,10 @@ GenTreeCall* Compiler::fgMorphArgs(GenTreeCall* callNode)
 #ifdef _TARGET_ARM_
 
         bool passUsingIntRegs;
-
+#ifdef ARM_SOFTFP
+        passUsingFloatRegs = false;
+        passUsingIntRegs   =  (intArgRegNum < MAX_REG_ARG);
+#else
         if (lateArgsComputed)
```

Comment:
Instead of this change, change the following line, below:

passUsingFloatRegs = !callIsVararg && (isHfaArg || varTypeIsFloating(argx));

to:

passUsingFloatRegs = !callIsVararg && (isHfaArg || varTypeIsFloating(argx)) && !opts.compUseSoftFP;

that should be sufficient.